src/Helper/PaymentHelper.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace Drupal\os2forms_payment\Helper;
+
+use Drupal\Core\Http\RequestStack;
+use Drupal\Core\Site\Settings;
+
+/**
+ * Payment helper class.
+ */
+class PaymentHelper {
+
+  /**
+   * Secret key for Nets EasyPay integration.
+   *
+   * @var string
+   */
+  protected $secretKey;
+
+  /**
+   * Checkout key for Nets EasyPay integration.
+   *
+   * @var string
+   */
+  protected $checkoutKey;
+
+  /**
+   * Terms URL for Nets EasyPay integration.
+   *
+   * @var string
+   */
+  protected $termsUrl;
+
+  /**
+   * Callback URL for Nets EasyPay integration.
+   *
+   * @var string
+   */
+  protected $callbackUrl;
+
+  /**
+   * {@inheritDoc}
+   */
+  public function __construct(
+    private readonly RequestStack $requestStack
+  ) {
+    $this->secretKey = Settings::get('os2forms_payment_secret_key');
+    $this->checkoutKey = Settings::get('os2forms_payment_checkout_key');
+    $this->termsUrl = Settings::get('os2forms_payment_terms_url');
+    $this->callbackUrl = Settings::get('os2forms_payment_callback_url');
+
+  }
+
+  /**
+   * Hook on webform submission presave, modifying data before submission.
+   *
+   * @return void
+   *   Return
+   */
+  public function webformSubmissionPresave(mixed $submission) {
+    $submission_data = $submission->getData();
+    $webform_elements = $submission->getWebform()->getElementsDecoded();
```

Comment:
Use `getElementsDecodedAndFlattened` to avoid running into issues with container and other stuff that form builders may add to forms:

```suggestion
    $webform_elements = $submission->getWebform()->getElementsDecodedAndFlattened();
```

Reply:
I see. Didn't really catch the difference when i wrote this as they all seemed to be identical.